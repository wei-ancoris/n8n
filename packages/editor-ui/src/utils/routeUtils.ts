import type { RouteRecordRaw } from 'vue-router';
import { VIEWS } from '@/constants';
import { useSettingsStore } from '@/stores/settings.store';

export const getPathAsRegexPattern = (path: RouteRecordRaw['path']): RegExp =>
	new RegExp(path.replace(/:\w+/g, '[^/]+'));

export function getTemplatesRedirect(defaultRedirect: VIEWS[keyof VIEWS]) {
	const settingsStore = useSettingsStore();
	const isTemplatesEnabled: boolean = settingsStore.isTemplatesEnabled;
	if (!isTemplatesEnabled) {
		return { name: defaultRedirect || VIEWS.NOT_FOUND };
	}

	return false;
}
