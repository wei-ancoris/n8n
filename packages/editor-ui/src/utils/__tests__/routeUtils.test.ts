import { getPathAsRegexPattern } from '../routeUtils';

const routes = [
	'/credentials',
	'/workflows',
	'/workflow',
	'/workflow/new',
	'/workflow/:name/debug/:executionId',
	'/workflow/:name/executions',
	'/workflow/:name/executions/:executionId',
	'/workflow/:workflowId/history/:versionId?',
	'/workflows/templates/:id',
];

const routesAsRegExps = routes.map(getPathAsRegexPattern);

test('should match the routes', () => {
	expect(routesAsRegExps).toMatchSnapshot();
});
