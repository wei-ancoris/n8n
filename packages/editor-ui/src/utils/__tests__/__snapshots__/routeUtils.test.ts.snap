// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match the routes 1`] = `
[
  /\\^\\\\/credentials\\$/,
  /\\^\\\\/workflows\\$/,
  /\\^\\\\/workflow\\$/,
  /\\^\\\\/workflow\\\\/new\\$/,
  /\\^\\\\/workflow\\\\/\\(\\[\\^/\\]\\+\\)\\\\/debug\\\\/\\(\\[\\^/\\]\\+\\)\\$/,
  /\\^\\\\/workflow\\\\/\\(\\[\\^/\\]\\+\\)\\\\/executions\\$/,
  /\\^\\\\/workflow\\\\/\\(\\[\\^/\\]\\+\\)\\\\/executions\\\\/\\(\\[\\^/\\]\\+\\)\\$/,
  /\\^\\\\/workflow\\\\/\\(\\[\\^/\\]\\+\\)\\\\/history\\\\/\\(\\[\\^/\\]\\+\\)\\$/,
  /\\^\\\\/workflows\\\\/templates\\\\/\\(\\[\\^/\\]\\+\\)\\$/,
]
`;
